lang/py/src/avro/io.py

```diff
@@ -580,9 +711,29 @@ def read_data(self, writers_schema, readers_schema, decoder):
     elif writers_schema.type == 'string':
       return decoder.read_utf8()
     elif writers_schema.type == 'int':
-      return decoder.read_int()
+      if hasattr(writers_schema, 'logical_type'):
```

Comment:
Same comment as above. This can be included inside [here](https://github.com/Yelp/avro/blob/trunk/lang/py/src/avro/io.py#L591).